oauth2/oauth2-client/src/main/java/org/springframework/security/oauth2/client/OAuth2AuthorizedClientSessionDestroyedListener.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2002-2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.security.oauth2.client;
+
+import java.util.Objects;
+
+import org.springframework.context.ApplicationEvent;
+import org.springframework.context.ApplicationListener;
+import org.springframework.security.core.context.SecurityContext;
+import org.springframework.security.core.session.SessionDestroyedEvent;
+import org.springframework.security.oauth2.client.authentication.OAuth2AuthenticationToken;
+import org.springframework.util.Assert;
+
+/**
+ * {@link OAuth2AuthorizedClientSessionDestroyedListener} is in charge of removing the
+ * {@link org.springframework.security.oauth2.client.OAuth2AuthorizedClient} upon session
+ * destroyed.
+ * <p>
+ * Note: This class use together with the
+ * {@link org.springframework.security.web.session.HttpSessionEventPublisher}.
+ * </p>
+ * @author Kazuki Shimizu
+ * @since 5.0
+ */
+public class OAuth2AuthorizedClientSessionDestroyedListener
```

Comment:
Let's rename to `OAuth2AuthenticationSessionDestroyedListener`. The thinking behind this is that the `ApplicationListener` for `SessionDestroyedEvent`'s is solely responsible for cleaning up an `OAuth2AuthenticationToken` session. The one task it has now is it cleans up the `OAuth2AuthorizedClient`, however, it may have more tasks in the future as we build things out.

Reply:
fix via e8fcdc72ca4d7a740d718b5f4a890f62b6a1452a